src/backend/updates.rs

```diff
@@ -69,46 +65,31 @@ fn post_upgrade() {
         || spawn(State::finalize_upgrade()),
     );
 
-    post_upgrade_fixtures();
+    sync_post_upgrade_fixtures();
 
     // post upgrade logic goes here
-    // set_timer(Duration::from_millis(0), move || {
-    //     spawn(post_upgrade_fixtures());
-    // });
+    set_timer(Duration::from_millis(0), move || {
+        spawn(async_post_upgrade_fixtures());
+    });
 }
 
-fn post_upgrade_fixtures() {
-    let last_id = read(|state| state.next_post_id);
-    mutate(|state| {
-        // fix tests, otherwise the new assertions break them
-        if state.memory.posts.len() == 0 {
-            return;
-        }
-
-        let num_posts = Post::count(state);
+fn sync_post_upgrade_fixtures() {}
 
-        for post_id in 0..last_id {
-            Post::mutate(state, &post_id, |post| {
-                post.archived = false;
-                Ok(())
-            })
-            .unwrap();
-        }
-
-        // The noop mutation above should move all posts to the heap
-        state.memory.assert_segments(1);
-        assert_eq!(state.memory.posts.len(), 0);
-        assert_eq!(state.posts.len(), num_posts);
-
-        // Now we can set the allocation block size
-        state.memory.set_block_size(300);
-
-        ic_cdk::println!(
-            "all posts loaded into heap (instructions: {}B, current SM size: {}Mb)",
-            performance_counter(0) / 1000000000,
-            (ic_cdk::api::stable::stable_size() << 16) / 1024 / 1024
-        );
+async fn async_post_upgrade_fixtures() {
+    // Fill the root posts index.
+    mutate(|state| {
```

Comment:
Did you verify that this completes in 20B instructions?

Reply:
I measured on the local replica and since this query is similar to `check` which seems to pass even as a simply query, I was not worried about that in the upgrade. Moreover, I just added a new commit with the counter of instructions that will always print how much was spent.

The counter says the post-upgrade spent 39B instructions. I am confused. The budget for upgrades is 20? Why does it work on a local replica?

Reply:
Btw, are these budgets in any kind of official spec? Or where could I look them up?

Reply:
Ok I found this: https://6qfxa-ryaaa-aaaai-qbhsq-cai.icp0.io/#/post/472562 Now it makes sense.

I keep forgetting what the limits are: 2/20 or 20/200...

Reply:
Yeah, it is 5B (query) / 20B (update) / 200B (upgrade):
https://internetcomputer.org/docs/current/developer-docs/smart-contracts/maintain/resource-limits